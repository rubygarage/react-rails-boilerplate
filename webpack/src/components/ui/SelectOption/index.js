import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'components/ui/Icon'

export function SelectOption({ children, className, isFocused, onFocus, onSelect, option, option: { title, icon } }) {
  const handleMouseDown = (event) => {
    event.preventDefault()
    event.stopPropagation()
    onSelect(option, event)
  }

  const handleMouseEnter = (event) => {
    onFocus(option, event)
  }

  const handleMouseMove = (event) => {
    if (isFocused) return

    onFocus(option, event)
  }

  return (
    <div
      className={className}
      onMouseDown={handleMouseDown}
      onMouseEnter={handleMouseEnter}
      onMouseMove={handleMouseMove}
      title={title}
    >
      <div className="flex align-center">
        <Icon type={icon} className="font-20 mr-10" />
        <span className="select-value-text">{children}</span>
      </div>
    </div>
  )
}

SelectOption.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  isFocused: PropTypes.bool,
  onFocus: PropTypes.func,
  onSelect: PropTypes.func,
  option: PropTypes.object.isRequired
}

export default SelectOption
