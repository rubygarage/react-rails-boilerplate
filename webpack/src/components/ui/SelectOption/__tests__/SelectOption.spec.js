import React from 'react'
import { shallow } from 'enzyme'
import SelectOption from 'components/ui/SelectOption'

describe('<SelectOption />', () => {
  const defaultProps = {
    option: { title: 'Audio', icon: 'icon-audio-circle' }
  }

  it('renders select option with icon', () => {
    const component = shallow(
      <SelectOption {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })
})
