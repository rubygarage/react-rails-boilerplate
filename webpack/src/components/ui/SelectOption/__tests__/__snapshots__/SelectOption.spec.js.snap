// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectOption /> renders select option with icon 1`] = `
<div
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseMove={[Function]}
  title="Audio"
>
  <div
    className="flex align-center"
  >
    <Icon
      className="font-20 mr-10"
      type="icon-audio-circle"
    />
    <span
      className="select-value-text"
    />
  </div>
</div>
`;
